app/qml/PositionMarker.qml

```diff
@@ -11,7 +12,62 @@ Item {
     property QgsQuick.PositionKit positionKit
     property color baseColor: InputStyle.highlightColor
     property bool withAccuracy: true
+    property int interval: 200 // Interval of direction marker updates in ms
+    property real threshold: 3 // threshold used to minimized direction marker updates (otherwise shaking)
+    property real speedLimit: 15  // Over speed limit, directions depends on direction of movement
+    property real direction: -1
+    property real groundSpeed: -1
 
+    Timer {
+      interval: positionMarker.interval; running: true; repeat: true
+      onTriggered: {
+        positionMarker.groundSpeed = __inputUtils.groundSpeedFromSource(positionKit)
+        if (groundSpeed > 0) {
+          var newDirection = compass.reading && groundSpeed < speedLimit ? compass.reading.azimuth : positionKit.direction
+        } else {
+          newDirection = compass.reading ? compass.reading.azimuth  : -1
+        }
+
+        var deltaAbsDirection = Math.abs(positionMarker.direction - newDirection)
```

Comment:
This does not seem to handle wrapping around 360 degrees - e.g. angles 358 and 0 should give delta = 2